nextcord/automod.py

```diff
@@ -0,0 +1,351 @@
+    keyword_filters: Optional[List[str]]
+        The custom filter for this auto moderation rule. `None` if not set.
+    presets: Optional[List[:class:`KeywordPresetType`]]
+        The pre-set type of this auto moderation rule. `None` if not set.
+    actions: List[:class:`AutoModerationAction`]
+        The actions that this auto moderation rule will execute if triggered.
+    """
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: AutoModerationRulePayload):
+        self.id: int = int(data["id"])
+        self.guild: Guild = guild
+        self._state: ConnectionState = state
+        self._from_payload(data)
+
+    def __int__(self):
+        return self.id
+
+    def __repr__(self):
+        attrs = (
+            ("id", self.id),
+            ("guild", self.guild),
+            ("guild_id", self.guild_id),
+            ("name", self.name),
+            ("creator_id", self.creator_id),
+            ("event_type", self.event_type),
+            ("trigger_type", self.trigger_type),
+            ("enabled", self.enabled),
+            ("exempt_role_ids", self.exempt_role_ids),
+            ("exempt_channel_ids", self.exempt_channel_ids),
+            ("keyword_filters", self.keyword_filters),
+            ("presets", self.presets),
+            ("actions", self.actions),
+        )
+        inner = " ".join("%s=%r" % t for t in attrs)
+        return f"<AutoModerationRule {inner}>"
+
+    def _from_payload(self, data: AutoModerationRulePayload):
+        self.guild_id: int = int(data["guild_id"])
+        self.name: str = data["name"]
+        self.creator_id: int = int(data["creator_id"])
+        self.event_type: EventType = try_enum(EventType, data["event_type"])
+        self.trigger_type: TriggerType = try_enum(TriggerType, data["trigger_type"])  # same
+        self.enabled: bool = data["enabled"]
+        self.exempt_role_ids: List[int] = [int(exempt_role) for exempt_role in data["exempt_roles"]]
+        self.exempt_channel_ids: List[int] = [
+            int(exempt_channel) for exempt_channel in data["exempt_channels"]
+        ]
+        self.keyword_filters: Optional[List[str]] = None
+        self.presets: Optional[List[KeywordPresetType]] = None
+
+        self._unpack_trigger_metadata(data["trigger_metadata"])
+        self.actions: List[AutoModerationAction] = []
+
+        for action in data["actions"]:
+            self.actions.append(AutoModerationAction(data=action))
+
+    def _unpack_trigger_metadata(self, trigger_metadata: TriggerMetadataPayload):
+        if trigger_metadata.get("keyword_filter") is not None:
+            self.keyword_filters = trigger_metadata["keyword_filter"]  # type: ignore
+        if trigger_metadata.get("presets") is not None:
+            self.presets = [try_enum(KeywordPresetType, preset) for preset in trigger_metadata["presets"]]  # type: ignore -- pylint messed up somehow
+
+    async def delete(self):
+        """|coro|
+
+        Delete this auto moderation rule."""
```

Comment:
```suggestion
        Delete this auto moderation rule.
        """
```
does this need any `raises` block for common http exceptions

Reply:
imo no

Reply:
And why is that? A lot of async methods that talk to discord contain a raises block

Reply:
maybe